apm-agent-attach-cli/src/main/java/co/elastic/apm/attach/Utils.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.attach;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class Utils {
+
+    private static final String AGENT_BASE_DIR = "agent";
+
+    private static Path agentDir = null;
+
+    public static Path getTargetAgentDir(String agentVersion) throws IOException {
```

Comment:
Given that `getTargetAgentDir` is at most called once, there is no need to cache the `agentDir`, and if we call it many times with different versions, it would be confusing:
```
Path path1 = Utils.getTargetAgentDir("1.0.0"); // will return path to version 1.0.0 as expected
Path path2 = Utils.getTargetAgentDir("1.2.3"); // will return path to version 1.0.0
```

Thus, I would suggest to just remove the static field `agentDir` and always re-compute the path from the version.

Reply:
I'll change it so we a Utils instance per version